.github/workflows/go.yml

```
@@ -0,0 +1,75 @@
+      # Checkout repo
+      - uses: actions/checkout@v4
+
+      # Go setup
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.23.11"
+
+      # Java setup for Allure CLI
+      - name: Set up java
+        uses: actions/setup-java@v3
```

actionlint (1.7.7)

the runner of "actions/setup-java@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)